### PR TITLE
PROCESSINGS: RetrieveAllPostImpressions

### LINKED ISSUE #415: PROCESSINGS: RetrieveAllPostImpressions

### PR DESCRIPTION
**Pull request by Javlonbek-dev**
Closes #415

### CODE REVIEW

File: Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+            this.postImpressionServiceMock.Setup(service =>
+                service.RetrieveAllPostImpressions()).Throws(dependencyException);
+
+            //when
+            Action retrieveAllPostImpressionAction = () =>
+                this.postImpressionProcessingService.RetrieveAllPostImpressionsAsync();
+
+            //then
+            Assert.Throws<PostImpressionProcessingDependencyException>(
```
Comment (cjdutoit) [inline Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs:36]: Change this to:

```
PostImpressionProcessingDependencyException actualPostImpressionProcessingDependencyException = 
    Assert.Throws<PostImpressionProcessingDependencyException>(retrieveAllPostImpressionAction);
```

also do this on the other excptin test

File: Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                retrieveAllPostImpressionAction);
+
+            this.postImpressionServiceMock.Verify(service =>
```
Comment (cjdutoit) [inline Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs:39]: Add:

```
actualPostImpressionProcessingDependencyException.Should().BeEquivalentTo(expectedPostImpressionProcessingDependencyException);
```
and the same for the other test

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Some small corrections needed

File: Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs (hunk 2 above)
Comment (Javlonbek-dev) [reply Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs:43]: Done

File: Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs (hunk 1 above)
Comment (Javlonbek-dev) [reply Taarafo.Core.Tests.Unit/Services/Processings/PostImpressions/PostImpressionProcessingServiceTests.Exceptions.RetrieveAll.cs]: Done

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**